### PR TITLE
Update LICENSE File with subcomponents

### PR DESCRIPTION
**Pull request by zachgk**
## Description ##
This PR updates the main LICENSE file by adding missing subcomponents of the project, combining the BSD licenses to their own section, and removing several subcomponents that were deleted.

It updates the two whitelists for automatic license checks by removing files that should not be there (this may involve adding the appropriate headers to files). Explanations were also added for each element in the license_header.py whitelist. Lastly, automatic checking for clojure and pyx files was also enabled.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by zachgk** [comment]
@mxnet-label-bot add [Licenses]

File: readthedocs.yml [hunk 1]
```diff
@@ -1,3 +1,20 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```
Comment (aaronmarkham) [inline readthedocs.yml]: I don't even think we use this file...

**Comment by ddavydenko** [comment]
@frankfliu , could you please also take a look?

**Comment by aaronmarkham** [comment]
We need the following to be sync'd up across master and 1.4.x:
* #13793 - needs to be in 1.4.x
* #13797 - needs to be in master
* #13799 - needs to be in 1.4.x
* #13808 - needs to be in 1.4.x
* #13851 - needs to be in 1.4.x

**Comment by zachgk** [comment]
@lupesko This should be cherry-picked and I added a PR for it at https://github.com/apache/incubator-mxnet/pull/13856. 

@aaronmarkham 
- #13793 was already put in to v1.4.x (https://github.com/apache/incubator-mxnet/commit/24daa7d79455c0a380ea7813b81f8c7ddd69bb1d)
- #13808 was closed, not merged. It was fixed in a different PR
- I don't think #13797 is still needed in master

In summary, the only change still needed should be PR #13856

**Pull request merged**